lib/mongoProvider.js

```diff
@@ -103,7 +103,7 @@ MongoCollection.prototype.find = function (query, req) {
     return self.db().then(function (db) {
         return self.beforeFindListeners.fire(query, req).then(function () {
 
-            return q.ninvoke(db.collection(self.name).find(query), "toArray").then(function(res) {
+            return q.ninvoke(db.collection(self.name).find(query, req), "toArray").then(function(res) {
```

Comment:
Ah, I see what you mean.

Unfortunately the `req` param you used is dedicated for different purpose than projection. Its purpose is to propagate the original http request which carries the authenticated user through (https://github.com/jsreport/jsreport/issues/151).

Also unfortunately, this will have to be changed on many places at once. I created and issue to make it happen, but it will be quite some work I'm afraid. 
https://github.com/jsreport/jsreport/issues/230 

I'm afraid, if you need the projection now, you'll have to touch the underlying mongo object

```js
documentStore.collection('templates').db().then(function (db) {
  db.collection('templates').find({}, {name: true})
})
```